### PR TITLE
[SPARK-45003][PYTHON][DOCS] Refine docstring of `asc/desc`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr refine docstring of `asc/desc ` and add some new examples.

### Why are the changes needed?
To improve PySpark documentation

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass Github Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -670,50 +671,74 @@ def asc(col: "ColumnOrName") -> Column:
...
-        the column specifying the order.
+        The column specifying the sort order.
 
     Examples
     --------
-    Sort by the column 'id' in the descending order.
+    Example 1: Sort DataFrame by 'id' column in ascending order.
 
-    >>> df = spark.range(5)
-    >>> df = df.sort(desc("id"))
```
Comment (LuciferYang) [inline python/pyspark/sql/functions.py:685]: The original example is incorrect, it demonstrated a desc example within function asc.

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Review by yaooqinn — APPROVED** [review]

File: python/pyspark/sql/functions.py [hunk 2]
```diff
@@ -723,27 +750,67 @@ def desc(col: "ColumnOrName") -> Column:
...
+    | id|value|
+    +---+-----+
+    |  2|    C|
+    |  4|    B|
+    |  3|    A|
+    +---+-----+
+
+    Example 3: Combine `asc` with `desc` to sort by multiple columns.
+
+    >>> from pyspark.sql.functions import asc, desc
```
Comment (allisonwang-db) [inline python/pyspark/sql/functions.py:790]: cc @zhengruifeng should we in the future use this import style instead? `import pyspark.sql.functions as sf`?

Comment (zhengruifeng) [reply python/pyspark/sql/functions.py:790]: yes, @LuciferYang what about making a follow up to apply `import pyspark.sql.functions as sf` (to be consistent with the quick start example) ?

Comment (zhengruifeng) [reply python/pyspark/sql/functions.py:790]: On second thought, I will try to change all the imports in functions.py in one batch.
So no need to make a follow up :)

Comment (LuciferYang) [reply python/pyspark/sql/functions.py:790]: OK

**Pull request closed without merging**